gui/gui_state_container.cpp

```diff
@@ -0,0 +1,27 @@
+#include "gui_state_container.h"
+
+#include <QtCharts/QXYSeries>
+
+QT_CHARTS_USE_NAMESPACE
+
+void GuiStateContainer::update(QAbstractSeries *pressure_series,
+                               QAbstractSeries *flow_series,
+                               QAbstractSeries *tv_series) {
+  auto now = SteadyClock::now();
+
+  QVector<QPointF> pressure_points, flow_points, tv_points;
+  for (const auto &[time, controller_status] : GetControllerStatusHistory()) {
+    int neg_millis_ago = TimeAMinusB(time, now).count();
+    pressure_points.append(
+        QPointF(neg_millis_ago * 0.001,
+                controller_status.sensor_readings.pressure_cm_h2o));
+    flow_points.append(
+        QPointF(neg_millis_ago * 0.001,
+                controller_status.sensor_readings.flow_ml_per_min));
+    tv_points.append(QPointF(neg_millis_ago * 0.001,
+                             controller_status.sensor_readings.volume_ml));
+  }
+  qobject_cast<QXYSeries *>(pressure_series)->replace(pressure_points);
+  qobject_cast<QXYSeries *>(flow_series)->replace(flow_points);
+  qobject_cast<QXYSeries *>(tv_series)->replace(tv_points);
```

Comment:
Does this copy the points vectors?  Or is there even more Qt magic here, it's doing a `swap` or something?

Reply:
Under the hood it's an assignment operator. I *think* it should support move-assignment so I'm going to add that.